portal-backend-python/routers/test_application.py

```diff
@@ -48,48 +67,106 @@ def sample_user(test_session):
     return user
 
 
-@pytest.fixture
-def sample_application(test_session, sample_user, sample_form):
-    application = Application(
-        user_id=sample_user.id,
-        form_key=sample_form.form_key,
-        status=ApplicationStatus.PENDING,
-        submission_json={"name": "John Doe", "email": "john@example.com"},
-    )
-    test_session.add(application)
-    test_session.flush()
-    return application
-
-
-class TestGetApplication:
+class TestApplication:
     def test_get_application_success(
         self,
         setup_dependency_overrides,
         sample_user,
-        sample_application,
+        sample_form,
+        test_session,
     ):
+        application = Application(
+            user_id=sample_user.id,
+            form_key=sample_form.form_key,
+            status=ApplicationStatus.PENDING,
+            submission_json={"name": "John Doe", "email": "john@example.com"},
+        )
+        test_session.add(application)
+        test_session.flush()
+
         response = client.get("application", params={"form_key": "test_form"})
 
         assert response.status_code == 200
         data = GetApplicationResponse(**response.json())
-        assert data.id == sample_application.id
+        assert data.id == application.id
         assert data.status == ApplicationStatus.PENDING
         assert data.form_data == {"name": "John Doe", "email": "john@example.com"}
 
-    # TODO: make these tests actually test something
-    def test_get_application_user_not_found(self, setup_dependency_overrides):
-        response = client.get("application", params={"form_key": "test_form"})
-
-        assert response.status_code == 404
-
-    def test_get_application_not_found(
+    @mock_aws
+    def test_submit_application_success(
```

Comment:
i think the choice of doing an end-to-end test instead of having unit tests is good here 👍